iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java

```diff
@@ -254,6 +262,32 @@ public void commitAlterTable(org.apache.hadoop.hive.metastore.api.Table hmsTable
     }
   }
 
+  @Override
+  public void rollbackAlterTable(org.apache.hadoop.hive.metastore.api.Table hmsTable, EnvironmentContext context)
```

Comment:
If this is called for any alter table op, will we be able to recognise the operation type and delete the metadata dir only in case of a true migration (and not do it for an alter table rename column, a simple alter table setprop case, etc)?

Reply:
I've marked in the EnvironmentContext that we are in the middle of a migration, so no other alter operation type remove the metadata dir.